### ISSUE TITLE
Union does not require two or more type arguments

### ISSUE THREAD

**Issue by alicederyn** [microsoft/pyright#4511]
**Describe the bug**
Code like `B = Union[A]` is being flagged as an error by pyright:
```text
error: Union requires two or more type arguments
```
But this seems to be perfectly legal syntax according to the Python documentation at https://docs.python.org/3/library/typing.html#typing.Union:

> Unions of a single argument vanish

**To Reproduce**
Run pyright against this:
```python
from typing import Union
A = Union[int]
```

**Expected behavior**
No errors

**VS Code extension or command-line**
Verified both in VSCode extension (v1.1.24) and command-line.

**Additional context**
This appears to have been an intentional change, to catch users writing e.g. `Union[Foo | Bar]`; see https://github.com/microsoft/pyright/issues/2479. However, while redundant, that's still totally legal, and means what you would expect.

We see this pattern in some cases where a type is declared that *in future* will be a union of multiple types (specifically TypedDicts), but is currently only a single type. Putting in the `Union` signposts the intention (and, coincidentally, simplifies a code generator we have that uses this pattern, but that's obviously not a problem to fix).

Decided to open this issue to query this behaviour.

**Comment by erictraut** [comment]
This is legal syntax, but it has no valid semantic meaning. The error is there to tell the user they are probably doing something wrong — and have a misunderstanding of what `Union` means or how it should be used.

This is by design and not considered a bug. If you really want to use `Union` with a single argument, you could use `# type: ignore` or `# pyright: ignore` to suppress the diagnostic.

**Comment by alicederyn** [comment]
Thanks, Eric. Is there a way to disable this globally in the `pyproject.toml`?

**Comment by erictraut** [comment]
No, there's no way to disable this particular diagnostic other than on a per-line basis.

**Comment by erictraut** [comment]
> simplifies a code generator we have

I would think that it would be trivial to modify your code generator to emit a type alias without an illegal Union form if there is a single subtype involved.

**Comment by alicederyn** [comment]
> I would think that it would be trivial to modify your code generator to emit a type alias without an illegal Union form if there is a single subtype involved.

Absolutely. I raised this partially to have an issue to point to for that PR (also motivating my follow-up question about whether it can be disabled). Thanks for your time!
